src/tests/ftest/cart/utest/utest_protocol.c

```diff
@@ -0,0 +1,98 @@
+/*
+ * (C) Copyright 2023 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <cmocka.h>
+#include <cart/api.h>
+#include "gurt/debug.h"
+
+#include <stdio.h>
+#include <stdlib.h>
```

Comment:
You should add this test to utils/utest.yaml and ci/test_files_to_stash.txt

See https://github.com/daos-stack/daos/pull/12993

Reply:
You may also want to run utils/run_utest.py --test_filter=protocol --memcheck to ensure it's valgrind clean before submitting though

Reply:
actually looks like valgrind is complaining about something, have to look at it...

Reply:
leak is fixed in https://github.com/mercury-hpc/mercury/pull/706 but will have to update mercury to pick it up, disabling part of that test in the meantime

Reply:
I'd expect we're updating anyway so perhaps this PR would be best waiting until that upgrade has landed, then we can land this one with all tests enabled.

Reply:
I'm not planning to push a new mercury update right at the moment so I would still prefer to get this merged so that at least @kjacque has a chance to experiment with the refactoring she wanted to do using those APIs. (And I would not expect her to need to pass the type of string that was causing a leak as she would just most certainly pass `NULL` as the input argument to get the list of all available transports)